openvdb/tools/FastSweeping.h

```diff
@@ -0,0 +1,1580 @@
+        mgr.foreachBottomUp(*this);//multi-threaded
+        tree.root().setBackground(std::numeric_limits<SdfValueT>::max(), false);
+        if (hasActiveTiles) tree.voxelizeActiveTiles();//multi-threaded
+
+        // cache the leaf node origins for fast lookup in the sweeping kernels
+
+        mParent->computeSweepMaskLeafOrigins();
+    }// FastSweeping::InitSdf::run
+
+    void operator()(const LeafRange& r) const
+    {
+        SweepMaskAccT sweepMaskAcc(mParent->mSweepMask);
+        math::GradStencil<SdfGridT, false> stencil(*mSdfGrid);
+        const SdfValueT isoValue = mIsoValue, above = mAboveSign*std::numeric_limits<SdfValueT>::max();//local copy
+        const SdfValueT h = mAboveSign*static_cast<SdfValueT>(mSdfGrid->voxelSize()[0]);//Voxel size
+        for (auto leafIter = r.begin(); leafIter; ++leafIter) {
+            SdfValueT* sdf = leafIter.buffer(1).data();
+            for (auto voxelIter = leafIter->beginValueAll(); voxelIter; ++voxelIter) {
+                const SdfValueT value = *voxelIter;
+                const bool isAbove = value > isoValue;
+                if (!voxelIter.isValueOn()) {// inactive voxels
+                  sdf[voxelIter.pos()] = isAbove ? above : -above;
+                } else {// active voxels
+                  const Coord ijk = voxelIter.getCoord();
+                  stencil.moveTo(ijk, value);
+                  const auto mask = stencil.intersectionMask( isoValue );
+                  if (mask.none()) {// most common case
+                    sdf[voxelIter.pos()] = isAbove ? above : -above;
+                  } else {// compute distance to iso-surface
+                    // disable boundary voxels from the mask tree
+                    sweepMaskAcc.setValueOff(ijk);
+                    const SdfValueT delta = value - isoValue;//offset relative to iso-value
+                    if (math::isApproxZero(delta)) {//voxel is on the iso-surface
+                      sdf[voxelIter.pos()] = 0;
+                    } else {//voxel is neighboring the iso-surface
+                      SdfValueT sum = 0;
+                      for (int i=0; i<6;) {
+                        SdfValueT d = std::numeric_limits<SdfValueT>::max(), d2;
+                        if (mask.test(i++)) d = math::Abs(delta/(value-stencil.getValue(i)));
+                        if (mask.test(i++)) {
+                          d2 = math::Abs(delta/(value-stencil.getValue(i)));
+                          if (d2 < d) d = d2;
+                        }
+                        if (d < std::numeric_limits<SdfValueT>::max()) sum += 1/(d*d);
+                      }
+                      sdf[voxelIter.pos()] = isAbove ? h / math::Sqrt(sum) : -h / math::Sqrt(sum);
+                    }// voxel is neighboring the iso-surface
+                  }// intersecting voxels
+                }// active voxels
+            }// loop over voxels
+        }// loop over leaf nodes
```

Comment:
Small style comment, but could we make the intention consistently 4 spaces?

Reply:
resolved - now please approve so I can go on vacation